client/src/main/kotlin/io/spine/examples/shareaware/client/Menu.kt

```diff
@@ -26,41 +26,94 @@
 
 package io.spine.examples.shareaware.client
 
+import androidx.compose.foundation.layout.Spacer
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.width
-import androidx.compose.material.Icon
-import androidx.compose.material.NavigationRail
-import androidx.compose.material.NavigationRailItem
-import androidx.compose.material.Text
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.NavigationRail
+import androidx.compose.material3.NavigationRailItem
+import androidx.compose.material3.NavigationRailItemDefaults
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.draw.clip
-import androidx.compose.ui.graphics.RectangleShape
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
 
 /**
  * Represents the item of the menu.
  */
-data class MenuItem(val name: String,
-                    val iconPath: String,
-                    val toPage: () -> Unit)
+public data class MenuItem(val name: String,
+                           val iconPath: String,
+                           val toPage: () -> Unit)
+
+/**
+ * Provides menu state and configuration.
+ */
+public object MenuModel {
+    public val currentPage: StateFlow<Pages> = CurrentPage.asStateFlow()
+    public val items: Map<Pages, MenuItem> = mapOf(
+        Pages.HOME to MenuItem(
+            "Home",
+            Icons.HOME
+        ) { CurrentPage.changesTo(Pages.HOME) },
+        Pages.WALLET to MenuItem(
+            "Wallet",
+            Icons.WALLET
+        ) { CurrentPage.changesTo(Pages.WALLET) },
+        Pages.MARKET to MenuItem(
+            "Market",
+            Icons.MARKET
+        ) { CurrentPage.changesTo(Pages.MARKET) },
+        Pages.INVESTMENTS to MenuItem(
+            "Investments",
+            Icons.INVESTMENT
+        ) { CurrentPage.changesTo(Pages.INVESTMENTS) },
+        Pages.WATCHLISTS to MenuItem(
+            "Watchlists",
+            Icons.WATCHLIST
+        ) { CurrentPage.changesTo(Pages.WATCHLISTS) }
+    )
+}
+
+/**
+ * The current page of the application.
+ */
+private object CurrentPage {
+    private val currentPage: MutableStateFlow<Pages> = MutableStateFlow(Pages.HOME)
+
+    /**
+     * Changes current page.
+     */
+    fun changesTo(page: Pages) {
```

Comment:
This signature looks weird. 

How can we improve?

1. Eliminate `Pages` and join its forces with `CurrentPage`, making a single type called `Page`.
2. Expose the current page via `Page` API.